esphome/components/hydreon_rgxx/hydreon_rgxx.cpp

```diff
@@ -195,7 +195,7 @@ void HydreonRGxxComponent::process_line_() {
       if (n == std::string::npos) {
         continue;
       }
-      int data = strtol(this->buffer_.substr(n + strlen(PROTOCOL_NAMES[i])).c_str(), nullptr, 10);
+      double data = strtof(this->buffer_.substr(n + strlen(PROTOCOL_NAMES[i])).c_str(), nullptr);
```

Comment:
We have a helper in ESPHome for number parsing (which ultimately uses `strtof`), but this way should make it easier on others taking this code as an example
```suggestion
      float data = parse_number<float>(this->buffer_.substr(n + strlen(PROTOCOL_NAMES[i]))).value();
```

Reply:
does not work, this is what I get:

```
[08:48:26][D][sensor:124]: 'Rain': Sending state 26038240.00000 mm with 4 decimals of accuracy
[08:48:26][D][hydreon_rgxx.sensor:200]: Received Acc: 26038240.000000
[08:48:26][D][sensor:124]: 'Rain Intensity': Sending state 26038240.00000 mm/h with 4 decimals of accuracy
[08:48:26][D][hydreon_rgxx.sensor:200]: Received RInt: 26038240.000000
[08:48:26][D][sensor:124]: 'Rain Event': Sending state 26038240.00000 mm with 4 decimals of accuracy
[08:48:26][D][hydreon_rgxx.sensor:200]: Received EventAcc: 26038240.000000
[08:48:26][D][sensor:124]: 'Rain Total': Sending state 26038240.00000 mm with 4 decimals of accuracy
[08:48:26][D][hydreon_rgxx.sensor:200]: Received TotalAcc: 26038240.000000
```

Unfortunately I am travelling at the moment and cannot debug.

Reply:
Strange that it does not work for you?
I loaded changes as [functionpointer](https://github.com/functionpointer) suggested in 
https://github.com/esphome/issues/issues/3248 and since then I get correct data from my RG-15